tests/test_parser.py

```diff
@@ -134,6 +135,115 @@ def test_parser_schema_violation():
         """,
     ]
 
+    # The validator reports a different error based on the python version, so
+    # we need to provide two copies for this to work.
+    if sys.version_info < (3, 7):
+        expected_errors.append(
+            """
+            ```
+            gleantest.event:
+              event_too_many_extras:
+                extra_keys:
+                  key_1:
+                    description: Sample extra key
+                  key_2:
+                    description: Sample extra key
+                  key_3:
+                    description: Sample extra key
+                  key_4:
+                    description: Sample extra key
+                  key_5:
+                    description: Sample extra key
+                  key_6:
+                    description: Sample extra key
+                  key_7:
+                    description: Sample extra key
+                  key_8:
+                    description: Sample extra key
+                  key_9:
+                    description: Sample extra key
+                  key_10:
+                    description: Sample extra key
+                  key_11:
+                    description: Sample extra key
+            ```
+
+            OrderedDict([('key_1', OrderedDict([('description', 'Sample extra
```

Comment:
Ugh.  This is ugly.  I filed a bug for this (doesn't need to be addressed as part of this PR): https://bugzilla.mozilla.org/show_bug.cgi?id=1639863

Reply:
Oh thanks!